### PR TITLE
Fixes #3273 : Improved edit custom theme

### LINKED ISSUE #3273: Bugs in custom theme dialog
**Actual Behaviour**

save button is neither disabled nor given a loader 
missing set background colour of chat text
Message Image URL does not gives preview of image

Screenshot 2020-01-13 at 5 54 27 PM
this is the util used to render message in chat box , It clearly shows a background colour necessity which was not passed

**Steps to Reproduce**

goto settings->theme->custom theme
try saving changes (you will notice no change in save button)
you will notice no option for messages background 
Also, image preview is not available

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by anishagg17**
Fixes #3273

Changes: 

- [X] Added circular loader to save button
- [x] Added option to change chat messages background colour
- [X]  Image URL shows preview of image

Demo Link: https://pr-3275-fossasia-susi-web-chat.surge.sh

Screenshots of the change:

**Changed  chat messages background colour**

Hnet com-image

Screenshot 2020-01-14 at 9 38 41 PM

**Image URL shows preview of image**

Screenshot 2020-01-13 at 6 47 54 PM

### CODE REVIEW

**Comment by anishagg17** [comment]
@akshatnitd please review this pr

File: src/components/ChatApp/MessageListItem/MessageListItem.react.js [hunk 1]
```diff
@@ -96,7 +97,9 @@ class MessageListItem extends React.Component {
       latestMessage,
       userGeoData,
       scrollBottom,
+      customThemeValue,
     } = this.props;
+    // console.log(customThemeValue);
```
Comment (akshatnitd) [inline src/components/ChatApp/MessageListItem/MessageListItem.react.js]: Remove comment

File: src/components/ChatApp/MessageListItem/MessageListItem.react.js [hunk 2]
```diff
@@ -118,18 +121,26 @@ class MessageListItem extends React.Component {
...
       this.state.showModal,
       this.onCloseModal,
+      customThemeValue,
     );
   }
 }
 
 function mapStateToProps(store) {
-  const { speechRate, speechPitch, ttsLanguage } = store.settings;
+  // console.log(store.settings);
```
Comment (akshatnitd) [inline src/components/ChatApp/MessageListItem/MessageListItem.react.js]: Remove comment

File: src/components/ChatApp/MessageListItem/generateMessageBubbleUtil.js [hunk 3]
```diff
@@ -386,12 +391,17 @@ export const generateMessageBubble = (
...
   showModal,
   onCloseModal,
+  customThemeValue,
 ) => {
   if (message && message.type === 'date') {
     return generateDateBubble(message);
   }
-
   const stringWithLinks = message ? message.text : '';
+  // console.log(customThemeValue);
```
Comment (akshatnitd) [inline src/components/ChatApp/MessageListItem/generateMessageBubbleUtil.js]: Remove comment

File: src/components/ChatApp/MessageListItem/generateMessageBubbleUtil.js [hunk 4]
```diff
@@ -688,10 +699,15 @@ export const generateMessageBubble = (
       </div>
     );
   }
-
+  const backgroundColor =
+    message.authorName === 'SUSI' ? susiBackgroundColor : userBackgroundColor;
+  // console.log(message);
```
Comment (akshatnitd) [inline src/components/ChatApp/MessageListItem/generateMessageBubbleUtil.js]: Remove comment

File: src/utils/colors.js [hunk 5]
```diff
@@ -33,13 +42,16 @@ export const getCustomThemeColors = ({ theme, customThemeValue }) => {
...
   }
+
   return {
     header,
     pane,
     body,
     composer,
     button,
     textarea,
+    susiBackgroundColor,
```
Comment (akshatnitd) [inline src/utils/colors.js]: Rename to `susiMessageBackgroundColor` and `userMessageBackgroundColor` everywhere.

Comment (anishagg17) [reply src/utils/colors.js]: Changes have been Done

**Pull request merged**